src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/common/SearchRequestDtoResolver.java

```diff
@@ -0,0 +1,45 @@
+package org.ibp.api.java.impl.middleware.inventory.manager.common;
+
+import org.generationcp.middleware.domain.inventory.manager.LotsSearchDto;
+import org.generationcp.middleware.domain.inventory.manager.SearchCompositeDto;
+import org.generationcp.middleware.domain.inventory.manager.TransactionsSearchDto;
+import org.generationcp.middleware.manager.api.SearchRequestService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+
+@Component
+public class SearchRequestDtoResolver {
+
+	@Autowired
+	private SearchRequestService searchRequestService;
+
+	public LotsSearchDto getLotsSearchDto(
+		final SearchCompositeDto searchCompositeDto) {
+
+		final LotsSearchDto searchDTO;
+		if (searchCompositeDto.getSearchRequestId() != null) {
+			searchDTO =
+				(LotsSearchDto) this.searchRequestService.getSearchRequest(searchCompositeDto.getSearchRequestId(), LotsSearchDto.class);
+		} else {
+			searchDTO = new LotsSearchDto();
+			searchDTO.setLotIds(new ArrayList<>(searchCompositeDto.getItemIds()));
+		}
+		return searchDTO;
+	}
```

Comment:
I wonder if something like this is possible. But we'll the dtos to implement a `setItemIds` 
```java
	public T getLotsSearchDto(final SearchCompositeDto searchCompositeDto) {

		final T searchDTO;
		if (searchCompositeDto.getSearchRequestId() != null) {
			searchDTO = (T) this.searchRequestService.getSearchRequest(searchCompositeDto.getSearchRequestId(), LotsSearchDto.class);
		} else {
			searchDTO = new T();
			searchDTO.setItemIds(new ArrayList<>(searchCompositeDto.getItemIds()));
		}
		return searchDTO;
	}
```